renku/ui/cli/dataset.py

```diff
@@ -1135,23 +1167,39 @@ def get_dataset_files(records):
             return tabulate(collection=records, columns="path,dataset,external", columns_mapping=columns)
 
         if not datasets and not dataset_files:
-            click.secho("Everything is up-to-date", fg=color.GREEN)
+            if not plain:
+                click.secho("Everything is up-to-date", fg=color.GREEN)
             return
 
         if datasets:
-            names = "\n\t".join(sorted([d.name for d in datasets]))
-            click.echo(f"The following imported datasets will be updated:\n\t{names}\n")
+            ds_names = sorted([d.name for d in datasets])
+            if plain:
+                ds_names = [f"d {n}" for n in ds_names]
+                click.echo(os.linesep.join(ds_names) + os.linesep)
+            else:
+                names = "\n\t".join(ds_names)
+                click.echo(f"The following imported datasets will be updated:\n\t{names}\n")
 
         if not dataset_files:
             return
 
         files = [f for f in dataset_files if not f.deleted]
         if files:
-            files = get_dataset_files(files)
-            click.echo(f"The following files will be updated:\n\n{files}\n")
+            if plain:
+                files = [f"f {f.path} {f.dataset.name}" for f in files]
+                click.echo(os.linesep.join(files) + os.linesep)
+            else:
+                files = get_dataset_files(files)
+                click.echo(f"The following files will be updated:\n\n{files}\n")
 
         deleted_files = [f for f in dataset_files if f.deleted]
         if deleted_files:
-            files = get_dataset_files(deleted_files)
-            message = " (pass '--delete' to remove them from datasets' metadata)" if not delete else ""
-            click.echo(f"The following files will be deleted{message}:\n\n{files}\n")
+            if plain:
+                files = [f"r {f.path} {f.dataset.name}" for f in deleted_files]
+                click.echo(os.linesep.join(files))
+            else:
+                files = get_dataset_files(deleted_files)
+                message = " (pass '--delete' to remove them from datasets' metadata)" if not delete else ""
+                click.echo(f"The following files will be deleted{message}:\n\n{files}\n")
+
+        ctx.exit(1)
```

Comment:
Just making sure this `ctx.exit(1)` is intentional and not leftover from troubleshooting/testing.

Reply:
Yes, if datasets are not up to date, it'll raise the status. That's what enables using it in e.g. the pre-commit hook